include/ccf/indexing/strategy.h

```diff
@@ -29,13 +29,10 @@ namespace ccf::indexing
 
     virtual void tick() {}
 
-    // Returns highest tx ID for which this index should be populated, or
-    // nullopt if it wants all Txs. Allows indexes to be populated
-    // lazily on-demand.
-    virtual std::optional<ccf::TxID> highest_requested()
-    {
-      return std::nullopt;
-    }
+    // Returns next tx for which this index should be populated, or
+    // nullopt if it wants none. Allows indexes to be populated
+    // lazily on-demand, or out-of-order, or reset
+    virtual std::optional<ccf::SeqNo> next_requested() = 0;
```

Comment:
If the `_Bucketed` strategy asks for a file (which it believes it wrote, and encrypted), and that file is missing, or corrupted, what should it do? We could be strict, view this as a malicious host, and fail hard, but I think that makes us too susceptible to accidental file mangling. The current behaviour is to happily say "no harm, no foul", and begin re-indexing from scratch. To re-index, we need to change the API between the indexer and Strategies - rather than assuming they're done with a Tx once they've seen it once, we ask each for the SeqNo they'd like next. This lets them reset to 0 to re-index.